src/thor/bidirectional_astar.cc

```diff
@@ -126,10 +126,9 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
     return false;
   }
 
-  // Update the time information
+  // Update the time information even if time is invariant to account for timezones
   auto offset_time =
-      from_transition ? time_info
-                      : time_info.forward(pred.cost().secs, static_cast<int>(nodeinfo->timezone()));
+      from_transition ? time_info : time_info.forward(0.f, static_cast<int>(nodeinfo->timezone()));
```

Comment:
previously it looked like we were varying the time, but there were no code paths that could get here where `time_info` would be `valid` and when its not valid then the `forward` function just exits immediately. ie this was a no-op before. now instead of varying the time we simply say `0.f` offset,  meaning same time just account for any changes in timezone.

in the future when bidirectional a* is time dependent and indeed varies the time as the route progresses we'll revert this change

Reply:
Why make the change if the previous code was a no-op and the change should be reverted later?

Reply:
because the code wont work if we dont :smile: 

i could change it to this to make it future proof:

```c++
auto offset_time =
￼      from_transition ? time_info : time_info.forward(invariant ? 0.f, pred.cost().secs, static_cast<int>(nodeinfo->timezone()));
```

Reply:
> i could change it to this to make it future proof

This sounds good I think. At least it signals to the reader that this functionality definitely depends on `invariant`